sdk/src/lib.rs

```diff
@@ -445,6 +454,23 @@ impl WorkflowHalf {
                 .send(activation)
                 .expect("Workflow should exist if we're sending it an activation");
         } else {
+            // When we failed to start a workflow, we never inserted it into the cache.
+            // But core sends us a `RemoveFromCache` job when we mark the StartWorkflow workflow activation
+            // as a failure, which we need to complete.
+            // Other SDKs add the workflow to the cache even when the workflow type is unknown/not found.
+            // To circumvent this, we simply mark any RemoveFromCache job for workflows that are not in the cache as complete.
+            if activation
+                .jobs
+                .iter()
+                .any(|j| matches!(j.variant, Some(Variant::RemoveFromCache(_))))
```

Comment:
This should check that the eviction is the *only* job, which should really be the only possible situation here.